### PR TITLE
partially update image copy operation tests; assorted small refactors

### PR DESCRIPTION
**Pull request by kainino0x**
These aren't close to passing yet, but this is a best-effort to partially update them.

-----

- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] Incomplete tests, if any, are marked with TODO or `.unimplemented()`.

**[Review requirements](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) are satisfied for:**

- [x] WebGPU readability
- [x] TypeScript readability

### CODE REVIEW

File: src/webgpu/api/operation/copyBetweenLinearDataAndTexture.spec.ts [hunk 1]
```diff
@@ -946,9 +958,9 @@ g.test('copy_various_mip_levels')
           mipLevel: 6,
         },
       ])
-      .combine(poptions('format', kSizedTextureFormats))
-      .filter(formatCanBeTested)
-      .expand(textureSizeExpander)
+      .expand(({ mipLevel, _texturePhysicalSizeAtMipLevelInBlocks }) =>
+        textureSizeExpander({ mipLevel, _texturePhysicalSizeAtMipLevelInBlocks, format: p.format })
```
Comment (Kangz) [inline src/webgpu/api/operation/copyBetweenLinearDataAndTexture.spec.ts:962]: Was this change needed because of the case/subcase change? Maybe textureSizeExpander could just be `computeTestTextureSizes` since it is no longer a functor.

File: docs/helper_index.md [hunk 2]
```diff
@@ -26,6 +26,3 @@ Generally, see:
 - [`ShaderValidationTest`](../src/webgpu/shader/validation/shader_validation_test.ts)
     - `expectCompileResult` Allows checking for compile success/failure, or failure with a
       particular error substring.
-- [`util/texture/image_copy.ts`](../src/webgpu/util/texture/image_copy.ts)
```
Comment (Kangz) [inline docs/helper_index.md]: Why is this removed?

File: src/webgpu/capability_info.ts [hunk 3]
```diff
@@ -146,6 +146,11 @@ export const kCompressedTextureFormatInfo = /* prettier-ignore */ makeTable(kTex
   'bc7-rgba-unorm-srgb':  [            ,        ,        ,          ,          ,          ,          ,              16,            4,             4, 'texture-compression-bc'],
 } as const);
 
+export type RegularTextureFormat = keyof typeof kRegularTextureFormatInfo;
```
Comment (Kangz) [inline src/webgpu/capability_info.ts:149]: nit: unrelated change?

File: src/webgpu/gpu_test.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { attemptGarbageCollection } from '../common/framework/util/collect_garba
 import { assert } from '../common/framework/util/util.js';
 
 import { EncodableTextureFormat, SizedTextureFormat } from './capability_info.js';
+import { makeBufferWithContents } from './util/buffer.js';
```
Comment (Kangz) [inline src/webgpu/gpu_test.ts:6]: nit: unrelated change?

File: src/webgpu/util/texture/image_copy.ts [hunk 5]
```diff
@@ -20,11 +21,13 @@ export function bytesInACompleteRow(copyWidth: number, format: SizedTextureForma
  * guess assuming `bytesPerRow` and `rowsPerImage` should be optimal.
  */
 export function dataBytesForCopy(
-  layout: GPUTextureDataLayout,
+  layout: GPUImageDataLayout,
```
Comment (Kangz) [inline src/webgpu/util/texture/image_copy.ts:24]: nit: could be integrated in the earlier commit.

File: src/webgpu/util/unions.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export function makeExtentDict(v: GPUExtent3D): GPUExtent3DDict {
```
Comment (Kangz) [inline src/webgpu/util/unions.ts:1]: normalizeExtent3D maybe?

**Review by Kangz — APPROVED** [review]
LGTM overall. I really wish Github allowed working on stacks of commits like Gerrit does.

File: docs/helper_index.md (hunk 2 above)
Comment (kainino0x) [reply docs/helper_index.md]: rebase mistake :)

File: src/webgpu/api/operation/copyBetweenLinearDataAndTexture.spec.ts (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/api/operation/copyBetweenLinearDataAndTexture.spec.ts:962]: Done. Though one thing I learned from this patch I'd like if it could become a functor again but it would be complex (requires passing the cases() object into subcases() which means making .combine() return a new object instead of mutating a builder)
